src/ops_generator.py

```diff
@@ -0,0 +1,70 @@
+import enum
+from threading import Thread, Event
+import logging
+import random
+import messages
+import copy
+
+from op import Op
+
+from threading import Thread
+from time import sleep
+from typing import List
+from nodes import SingleItemNode, NodeState
+import operations
+
+log = logging.getLogger()
+
+class OpsGenerator(Thread):
+    def __init__(self, node_process, cluster, queue, delay=1, failure_rate=0.2, recover_rate=0.2):
+        '''
+            Random Operation Generator
+        '''
+        super(OpsGenerator, self).__init__()
+        self.name = "ops-generator-{}".format(node_process.node.get_id())
+        self.node_process = node_process
+        self.node = node_process.node
+        self.cluster = cluster
+        self.queue = queue
+        self.delay = delay
+        self.state_helper = node_process.state_helper
+        self.failure_rate = failure_rate
+        self.recover_rate = recover_rate
+        self.running = Event()
+        self.running.set()  # set the stage to run by default
+
+    def getRandomOperation(self):
+        if not self.is_node_part_of_flow(node_id=self.node.node_id):
+            return None
+
+        if self.node.state == NodeState.active and random.random() < self.failure_rate:
+            return Op.Kill
+        elif self.node.state == NodeState.inactive and random.random() < self.recover_rate:
+            return Op.Recover
+
+        return None
```

Comment:
how will this guarantee that the failure rate is, for example, 2 nodes per minute? It still seems random to me. We want deterministic crashes. So a thread can select x% random node ids and then crash/stop each node as it wishes.